test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/AbstractService.java

```diff
@@ -195,8 +236,30 @@ public String getID() {
                 .build();
     }
 
+    private Map<String, Object> getResources(String limitsCpu, String limitsMem, String requestsCpu, String requestsMem) {
+        return ImmutableMap.<String, Object>builder()
+                .put("limits", ImmutableMap.builder()
+                        .put("cpu", limitsCpu)
+                        .put("memory", limitsMem)
+                        .build())
+                .put("requests", ImmutableMap.builder()
+                        .put("cpu", requestsCpu)
+                        .put("memory", requestsMem)
+                        .build())
+                .build();
+    }
+
     private V1beta1CustomResourceDefinition getPravegaCRD() {
 
+        String pravegaOperatorTag = PRAVEGA_OPERATOR_IMAGE.substring(PRAVEGA_OPERATOR_IMAGE.lastIndexOf(":") + 1);
```

Comment:
Why are we doing this inside getPravegaCRD?
Please create a separate method from this code and invoke it @ln 95, where isLaterOperator is defined. That way isLaterOperator can be final.

Reply:
done